File: anagram.py
#!/usr/bin/env python3
import sys
anagram = {}
with open(sys.argv[1], encoding='utf8') as ifs:
    for line in ifs:
        line = line.strip().lower()
        s = ''.join(sorted(line))
        anagram.setdefault(s, []).append(line)
for k, v in anagram.items():
    if len(v) > 1:
        print(" ".join(v))

File: anagram.pl
#!/usr/bin/env perl
open(my $ifs, "<:utf8", $ARGV[0]) or die $!;
binmode STDOUT, ":utf8";
my %anagram;
foreach my $line (split "\n", do { local $/; lc(<$ifs>) }) {
	my $sorted = join '', sort split(//, $line);
	push @{$anagram{$sorted}}, $line;
}
foreach my $k (keys %anagram) {
	print join(" ", @{$anagram{$k}}), "\n" if (@{$anagram{$k}} > 1);
}
